swaybar/main.c

```diff
@@ -1034,6 +1067,44 @@ int i3json_handle_fd(int fd) {
 	return i3json_parse();
 }
 
+bool handle_ipc_event() {
+	struct ipc_response *resp = ipc_recv_response(ipc_event_socketfd);
+	switch (resp->type) {
+	case IPC_EVENT_WORKSPACE:
+		ipc_update_workspaces();
+		break;
+	case IPC_EVENT_MODE: ;
```

Comment:
Yeah it's strange, but It's because gcc complains with `error: a label can only be part of a statement and a declaration is not a statement` if you have a declaration right after a `case`. I can move `json_object *result;` up to before the `switch` if you like.